src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/ProvisionedInfrastructure.kt

```diff
@@ -0,0 +1,34 @@
+package com.atlassian.performance.tools.awsinfrastructure.api
+
+import com.atlassian.performance.tools.aws.api.Resource
+import com.atlassian.performance.tools.aws.api.UnallocatedResource
+import com.atlassian.performance.tools.infrastructure.api.virtualusers.VirtualUsers
+
+class ProvisionedInfrastructure<out T : VirtualUsers>
+@Deprecated("Use ProvisionedInfrastructure.Builder instead.")
+constructor(
+    val infrastructure: Infrastructure<T>,
+    val resource: Resource
+) {
+    object Defaults {
```

Comment:
New API, is it necessary?

Reply:
It's not necessary, but it covers a lib consumer use case where it doesn't want to reconstruct a default submodule, but it wants to append it's functionality, e.g. wants to add additional resources to be released with the `Resource` class or wants to grant more permissions via `AccessProvider` or simply wants to monitor it's usages to print the diagnostics in the log or other persistance layer.

Reply:
Example: For `DataCenterFormula` you may want add more to `AccessRequester` which you can specify in the `DataCenterFormula.Builder`. You can do that by wrapping the `DataCenterFormula.Defaults.accessRequester` in `MultiAccessRequester` + your access. Otherwise you'd need to dig into the code and base your implementation on the internal implementation of the `DataCenterFormula`.

Reply:
Good use case. We should log this change.

Reply:
- [ ] Add `Defauts` API to the changelog